### PR TITLE
Move CustomerReturn logic to ReturnAuthorization model

### PR DESCRIPTION
**Pull request by aleph1ow**
Due to a previous bad interpretation of solidus return flow creation we moved the
return creation from CustomerReturn to ReturnAuthorization model.
CustomerReturn will be implied when a shipwire return will be marked as
received.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,4 @@ nbproject
 *.swp
 spec/dummy
 Gemfile.lock
+vendor
```
Comment (kennyadsl) [inline .gitignore:12]: ?

Comment (aleph1ow) [reply .gitignore:12]: Reproducing CI environments locally, to test the gem that fails with some solidus versions, makes this directory to be created.

Comment (kennyadsl) [reply .gitignore:12]: not sure to follow, when vendor directory is created?

File: solidus_shipwire.gemspec [hunk 2]
```diff
@@ -20,6 +20,8 @@ Gem::Specification.new do |s|
   s.add_dependency %q<solidus_core>, ['< 3', '>= 1.0.6']
   s.add_dependency %q<solidus_backend>, ['< 3', '>= 1.0.6']
   s.add_dependency %q<shipwire>, '~> 2.0'
+  
+  s.add_runtime_dependency 'solidus_support'
```
Comment (kennyadsl) [inline solidus_shipwire.gemspec:24]: why runtime dependency?

File: db/migrate/4_add_shipwire_id_to_customer_return.rb [hunk 3]
```diff
@@ -1,4 +1,4 @@
-class AddShipwireIdToCustomerReturn < ActiveRecord::Migration[4.2]
+class AddShipwireIdToCustomerReturn < SolidusSupport::Migration[4.2]
   def change
     add_column :spree_customer_returns, :shipwire_id, :integer, default: nil, null: true
```
Comment (kennyadsl) [inline db/migrate/4_add_shipwire_id_to_customer_return.rb:3]: is this `shipwire_id` on `CustomerReturns` still needed?

File: spec/models/spree/return_athorization_spec.rb [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+        it 'post to shipwire' do
+          expect(return_authorization).to receive(:process_shipwire_return!).and_call_original
+          return_authorization.save
+          expect(return_authorization.shipwire_id).not_to be nil
+        end
+
+        it 'places a single item in the return' do
+          expect(return_authorization.to_shipwire[:items]).to eq items_array
+        end
```
Comment (kennyadsl) [inline spec/models/spree/return_athorization_spec.rb:95]: should we test that the `return_authorization` has the correct `shipwire_id` as well?

**Review by kennyadsl — COMMENTED** [review]
just some questions

**Comment by DanielePalombo** [comment]
Close in favour of #17

**Pull request closed without merging**